### ISSUE TITLE
Support UTF-8 with BOM

### ISSUE THREAD

**Issue by aegoroff** [pelletier/go-toml#250]
**Is your feature request related to a problem? Please describe.**
Now the lib doesn't support toml files in UTF-8 encoding with BOM

**Describe the solution you'd like**
The lib correctly handles UTF-8 encoding with BOM files

**Comment by pelletier** [comment]
Hi! Do you have a file that exhibits this issue that we could test with?

**Comment by aegoroff** [comment]
Sure :) you can create any TOML file and save it in any text editor using UTF8 with BOM, for example Notepad++.
[copyto.txt](https://github.com/pelletier/go-toml/files/2596004/copyto.txt)
I attached one. You can use these types to Unmarshal
```
type tomlConfig struct {
	Title       string
	Sources     map[string]source
	Definitions map[string]definition
}

type source struct {
	Source string
}

type definition struct {
	SourceLink string
	Source     string
	Target     string
}
```

**Comment by pelletier** [comment]
@aegoroff should be fixed in 539dd095b346bc48c8d9348137b9535ac1efa133. Feel free to reopen if not!
